app/modules/tcs34725.c

```diff
@@ -0,0 +1,359 @@
+
+static bool							_tcs34725Initialised = false;
+static int32_t						_tcs34725SensorID = 0;
+static tcs34725Gain_t				_tcs34725Gain = TCS34725_GAIN_1X;
+static tcs34725IntegrationTime_t	_tcs34725IntegrationTime = TCS34725_INTEGRATIONTIME_2_4MS;
+
+os_timer_t tcs34725_timer; // timer for forced mode readout
+sint32_t cb_tcs_en;
+sint32_t self_ref;
+
+/**************************************************************************/
+/*!
+		@brief	Writes an 8 bit values over I2C
+*/
+/**************************************************************************/
+uint8_t tcs34725Write8 (uint8_t reg, uint8_t value)
+{
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(TCS34725_BUS_ID, TCS34725_COMMAND_BIT | reg );
+	platform_i2c_send_byte(TCS34725_BUS_ID, value);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Reads a 8 bit values over I2C
+*/
+/**************************************************************************/
+uint8_t tcs34725Read8(uint8_t reg)
+{
+	uint8_t value;
+	
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(TCS34725_BUS_ID, TCS34725_COMMAND_BIT | reg);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_RECEIVER);
+	value = platform_i2c_recv_byte(TCS34725_BUS_ID, 0);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	return value;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Reads a 16 bit values over I2C
+*/
+/**************************************************************************/
+uint16_t tcs34725Read16(uint8_t reg)
+{		
+	uint8_t low = tcs34725Read8(reg);
+	uint8_t high = tcs34725Read8(++reg);
+	
+	return (high << 8) | low;
+}
+/**************************************************************************/
+/*!
+		@brief	Finishes enabling the device
+*/
+/**************************************************************************/
+uint8_t tcs34725EnableDone(lua_State* L)
+{
+	dbg_printf("Enable finished\n");
+	os_timer_disarm (&tcs34725_timer);
+	tcs34725Write8(TCS34725_ENABLE, TCS34725_ENABLE_PON | TCS34725_ENABLE_AEN);
+
+	/* Ready to go ... set the initialised flag */
+	_tcs34725Initialised = true;
+	
+	/* This needs to take place after the initialisation flag! */
+	tcs34725SetIntegrationTime(TCS34725_INTEGRATIONTIME_2_4MS, L);
+	tcs34725SetGain(TCS34725_GAIN_60X, L);	
+	
+	lua_rawgeti(L, LUA_REGISTRYINDEX, cb_tcs_en); // Get the callback to call
+	luaL_unref(L, LUA_REGISTRYINDEX, cb_tcs_en); // Unregister the callback to avoid leak
+	lua_rawgeti(L, LUA_REGISTRYINDEX, self_ref);
```

Comment:
Remove `self_ref` as proposed below.
If you decide to keep `self_ref`, then tell `lua_call()` that there are **2** items available on the stack.